### DISCUSSION TITLE
Why is a 24-bit check used here instead of a 32-bit check?

### DISCUSSION THREAD

**Discussion by fumeboy** [eh-steve/goloader#7] [Q&A] [answered]
https://github.com/eh-steve/goloader/blob/9fbc78a6d5210680c45d9d54dedbaf66c1314e42/mmap/mmap_darwin_arm64.go#L50

**Comment by eh-steve** [comment, marked as answer]
Arm64v8 instructions don't use immediate 4 byte values next to the instructions like x86, instead the addresses/offsets are encoded as part of the 32-bit instruction so they generally have a maximum of 24 bits available for Branches (and 33-bits for ADRP+ADD relocs across 2 instructions)
